gm4_combat_expanded/data/gm4_combat_expanded/functions/armor/modifier/type/canine/wolf_remove_armor.mcfunction

```diff
@@ -0,0 +1,12 @@
+# remove armor from wolf
+# @s = wolf that is waering armor
+# at @s
+# run from armor/type/canine/wolf_interact_check
+
+# spawn item with the data
+data modify storage gm4_combat_expanded:temp body_armor_item.data set from entity @s body_armor_item
+function gm4_combat_expanded:armor/modifier/type/canine/eval_dropped_armor with storage gm4_combat_expanded:temp body_armor_item
```

Comment:
Instead of using a macro for this, you can summon the item and then use `item replace entity <item> contents from entity @s body_armor_item`